sdk/core/src/main/java/org/apache/mesos/config/ConfigurationComparator.java

```diff
@@ -0,0 +1,15 @@
+package org.apache.mesos.config;
+
+/**
+ * This interface defines a way to compare two {@link Configuration} objects for equality. This
+ * interface is intentionally separate from {@link Object#equals(Object)} in order to have very
+ * explicit requirements around equality checks when they're needed.
+ *
+ * @param <T> The {@code Configuration} type that will be parsed.
+ */
+public interface ConfigurationComparator<T extends Configuration> {
```

Comment:
Got it. Java's interface is a bit weird, there's `compare` and there's a `equals` which orverrides Object's `equals`. 
This will do for now. I was just trying to figure out if we can avoid duplication of interfaces.